include/exec/start_now.hpp

```diff
@@ -75,107 +75,143 @@ namespace exec {
           -> __operation<_StgRef, std::remove_cvref_t<_Receiver>> {
         return {static_cast<_Self&&>(__self).__stgref_, static_cast<_Receiver&&>(__rcvr)};
       }
 
-      template <class _Env>
-      auto get_completion_signatures(_Env&&) -> __completions_t<_Env> {
+      auto get_completion_signatures(stdexec::__ignore = {}) -> __completions_t {
         return {};
       }
     };
 
-    template <class _SenderId, class _EnvId>
+    template <class _EnvId, class _AsyncScopeId, class... _SenderId>
     struct __storage {
-      using _Sender = stdexec::__t<_SenderId>;
       using _Env = stdexec::__t<_EnvId>;
+      using _AsyncScope = stdexec::__t<_AsyncScopeId>;
       using __receiver_t = __receiver<__ref_t<__storage>>;
       using __sender_t = __sender<__ref_t<const __storage>>;
-      using __env_t = __env::__join_t<_Env, __env::__with<inplace_stop_token, get_stop_token_t>>;
-
-      static_assert(sender_to<_Sender, __receiver_t>, "The sender passed to start_now does not satisfy the constraints");
+      using __env_t = stdexec::__env::__join_t<stdexec::__env::__with<stdexec::inplace_stop_token, stdexec::get_stop_token_t>, _Env>;
+      template<class _S>
+      using __nested_t = decltype(stdexec::__declval<_AsyncScope&>().nest(stdexec::__declval<_S&&>()));
 
-      mutable std::atomic<const __joiner*> __joiner_{&__empty_joiner_};
+      mutable std::atomic<const __joiner*> __joiner_;
+      mutable std::atomic<int> __pending_;
 
       STDEXEC_ATTRIBUTE((no_unique_address))
-      inplace_stop_source source;
+      stdexec::inplace_stop_source __source_;
       STDEXEC_ATTRIBUTE((no_unique_address))
       __env_t __env_;
       STDEXEC_ATTRIBUTE((no_unique_address))
-      connect_result_t<_Sender, __receiver_t> __op_state_;
+      stdexec::__decayed_tuple<stdexec::connect_result_t<__nested_t<stdexec::__t<_SenderId>>, __receiver_t>...> __op_state_;
+
+      template<class _Sender>
+      auto __construct(_Sender&& __sndr) noexcept {
+        return [&, this](){return stdexec::connect(static_cast<_Sender&&>(__sndr), __receiver_t{__ref(*this)});};
+      }
 
-      __storage(_Sender&& __sndr, _Env __env)
-        : __env_(__env::__join(std::move(__env), __env::__with{source.get_token(), get_stop_token}))
-        , __op_state_(connect(static_cast<_Sender&&>(__sndr), __receiver_t{__ref(*this)})) {
-        start(__op_state_);
+      __storage(_Env __env, _AsyncScope& __scope, stdexec::__t<_SenderId>... __sndr)
+        : __joiner_(&__empty_joiner_)
+        , __pending_(0)
+        , __source_()
+        , __env_(stdexec::__env::__join(stdexec::__env::__with{__source_.get_token(), stdexec::get_stop_token}, std::move(__env)))
+        , __op_state_(stdexec::__conv{__construct(__scope.nest(__sndr))}...) {
+        __pending_ = sizeof...(_SenderId);
+        stdexec::__apply([](auto&... __op_state) noexcept { bool arr[]{(stdexec::start(__op_state), true)...}; (void)arr; }, __op_state_);
```

Comment:
you can fold over the comma operator like this:

```suggestion
        stdexec::__apply([](auto&... __op_state) noexcept { (stdexec::start(__op_state),...); }, __op_state_);
```